internal/core/project.go

```diff
@@ -24,47 +24,86 @@ import (
 type ProjectReport struct {
 	*limes.ProjectReport
 
+	HasRatesOnly bool
+
 	DomainID   string
 	DomainName string
 }
 
 // LimesProjectsToReportRenderer wraps the given limes.ProjectReport in a
 // ProjectReport and returns a []LimesReportRenderer.
-func LimesProjectsToReportRenderer(in []limes.ProjectReport, domainID, domainName string) []LimesReportRenderer {
+func LimesProjectsToReportRenderer(
+	in []limes.ProjectReport,
+	domainID, domainName string,
+	hasRatesOnly bool) []LimesReportRenderer {
+
 	out := make([]LimesReportRenderer, 0, len(in))
 	for _, rep := range in {
 		rep := rep
 		out = append(out, ProjectReport{
 			ProjectReport: &rep,
 			DomainID:      domainID,
 			DomainName:    domainName,
+			HasRatesOnly:  hasRatesOnly,
 		})
 	}
 	return out
 }
 
 var csvHeaderProjectDefault = []string{"domain id", "project id", "service", "resource", "quota", "usage", "unit"}
 
-var csvHeaderProjectLong = []string{"domain id", "domain name", "project id", "project name", "area", "service",
-	"category", "resource", "quota", "burst quota", "usage", "physical usage", "burst usage", "unit", "scraped at (UTC)"}
+var csvHeaderProjectLong = []string{
+	"domain id", "domain name", "project id", "project name", "area", "service",
+	"category", "resource", "quota", "burst quota", "usage", "physical usage", "burst usage", "unit", "scraped at (UTC)",
+}
 
 // GetHeaderRow implements the LimesReportRenderer interface.
-func (p ProjectReport) getHeaderRow(csvFmt CSVRecordFormat) []string {
-	switch csvFmt {
+func (p ProjectReport) getHeaderRow(opts *OutputOpts) []string {
+	if p.HasRatesOnly {
+		return p.getRatesHeaderRow(opts)
+	}
+
+	switch opts.CSVRecFmt {
 	case CSVRecordFormatLong:
 		return csvHeaderProjectLong
 	case CSVRecordFormatNames:
 		h := csvHeaderProjectDefault
-		h[0] = "domain name"
+		h[0] = domainName
 		h[1] = "project name"
 		return h
 	default:
+
 		return csvHeaderProjectDefault
 	}
 }
 
+var csvHeaderProjectRatesDefault = []string{"domain id", "project id", "service", "rate", "limit", "window", "usage as bigint", "unit"}
+
+var csvHeaderProjectRatesLong = []string{
+	"domain id", "domain name", "project id", "project name", "area", "service",
+	"rate", "limit", "default limit", "window", "default window", "usage as bigint", "unit", "scraped at (UTC)",
+}
```

Comment:
The "usage as bigint" field can be called just "usage". I named it `usageAsBigint` in the API to explain why it is a string field (JSON integers are limited to 64 bit). Since the CSV output is stringly-typed, this distinction is not necessary here.